packages/core/test/zodiac/OptimisticGovernor.js

```diff
@@ -1,691 +1,601 @@
+
+    await bondToken.methods.mint(proposer, doubleTotalBond).send({ from: owner });
+    await bondToken.methods.approve(optimisticOracleModule.options.address, doubleTotalBond).send({ from: proposer });
+    await bondToken.methods.mint(disputer, totalBond).send({ from: owner });
+    await bondToken.methods.approve(optimisticOracle.options.address, totalBond).send({ from: disputer });
+  });
+
+  it("Constructor validation", async function () {
+    // 0 liveness.
+    assert(
+      await didContractThrow(
+        OptimisticGovernor.new(
+          finder.options.address,
+          avatar.options.address,
+          bondToken.options.address,
+          bond,
+          rules,
+          identifier,
+          0,
+          timer.options.address
+        ).send({ from: owner })
+      )
+    );
+
+    // Unapproved token.
+    assert(
+      await didContractThrow(
+        OptimisticGovernor.new(
+          finder.options.address,
+          avatar.options.address,
+          (await ERC20.new("BOND", "BOND", 18).send({ from: owner })).options.address,
+          bond,
+          rules,
+          identifier,
+          liveness,
+          timer.options.address
+        ).send({ from: owner })
+      )
+    );
+
+    // Unapproved identifier.
+    assert(
+      await didContractThrow(
+        OptimisticGovernor.new(
+          finder.options.address,
+          avatar.options.address,
+          bondToken.options.address,
+          bond,
+          rules,
+          utf8ToHex("Unapproved"),
+          liveness,
+          timer.options.address
+        ).send({ from: owner })
+      )
+    );
+  });
+
+  it("Valid proposals should be hashed and stored and emit event", async function () {
+    // Issue some test tokens to the avatar address.
+    await testToken.methods.allocateTo(avatar.options.address, toWei("3")).send({ from: accounts[0] });
+    await testToken2.methods.allocateTo(avatar.options.address, toWei("2")).send({ from: accounts[0] });
+
+    // Construct the transaction data to send the newly minted tokens to proposer and another address.
+    const txnData1 = constructTransferTransaction(proposer, toWei("1"));
+    const txnData2 = constructTransferTransaction(rando, toWei("2"));
+    const txnData3 = constructTransferTransaction(proposer, toWei("2"));
+    const operation = 0; // 0 for call, 1 for delegatecall
+
+    // Send the proposal with multiple transactions.
+    const transactions = [
+      { to: testToken.options.address, operation, value: 0, data: txnData1 },
+      { to: testToken.options.address, operation, value: 0, data: txnData2 },
+      { to: testToken2.options.address, operation, value: 0, data: txnData3 },
+    ];
+
+    const explanation = utf8ToHex("These transactions were approved by majority vote on Snapshot.");
+
+    let receipt = await optimisticOracleModule.methods
+      .proposeTransactions(transactions, explanation)
+      .send({ from: proposer });
+
+    const { proposalHash } = (
```

Comment:
Should you also test that the OO is being requested for a price correctly?

Reply:
Good call, will add this in a follow-up.